### PR TITLE
Pmd clean up

### PR DESCRIPTION
**Pull request by manticore-projects**
This is a CLEAN PR which containts only Annotations to switch off some of the Codazy/PMD rules regarding the Deparsers and Visitors.
Also added some Java Code Formatter Configuration.

No code changes.

### CODE REVIEW

**Comment by manticore-projects** [comment]
Summary of that PR in one chart:
image

File: nb-configuration.xml [hunk 1]
```diff
@@ -20,5 +20,14 @@
         <com-junichi11-netbeans-changelf.use-global>false</com-junichi11-netbeans-changelf.use-global>
         <org-netbeans-modules-javascript2-requirejs.enabled>true</org-netbeans-modules-javascript2-requirejs.enabled>
         <netbeans.hint.jdkPlatform>JDK_1.8</netbeans.hint.jdkPlatform>
+        <netbeans.checkstyle.format>true</netbeans.checkstyle.format>
```
Comment (wumpz) [inline nb-configuration.xml:23]: Was necessary to introduce a project level formatting section? Why?

File: src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java [hunk 2]
```diff
@@ -244,6 +244,7 @@ public void visit(IntegerDivision division) {
 
     @Override
     public void visit(DoubleValue doubleValue) {
+    // intentional (commented) empty methods.
```
Comment (wumpz) [inline src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java:247]: Which PMD rule forces this?

Comment (wumpz) [reply src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java:247]: I would prefer it without any comment. The name itself is documentation enough.

**Review by wumpz — CHANGES_REQUESTED** [review]

File: src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java (hunk 2 above)
Comment (manticore-projects) [reply src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java:247]: The rule `<rule ref="category/java/documentation.xml/UncommentedEmptyMethodBody" />` is applied by Codacy and will reject any commit affecting this class.
Although I can replace it by a Class wide Annotation.

File: nb-configuration.xml (hunk 1 above)
Comment (manticore-projects) [reply nb-configuration.xml]: Removed.

File: src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java (hunk 2 above)
Comment (wumpz) [reply src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java]: Damned. You are right. However, I will remove it from Codacy.

**Comment by manticore-projects** [comment]
On Wed, 2021-06-02 at 05:14 -0700, Tobias wrote:
> Damned. You are right. However, I will remove it from Codacy.

Have a look at the project's PMD config. I marked the rules in palce
fro Codazy only and would like to suggest to remove these from Codazy.
It is not a bad thing at all, but some of the rules really create more
headache then helping.

File: src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java (hunk 2 above)
Comment (wumpz) [reply src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java]: Removed.

Comment (manticore-projects) [reply src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java]: Yes, but I have solved it before already.
I have removed these comments already and you just need to accept the PR pls.

**Pull request merged**